comp/autoscaling/datadogclient/impl/status.go

```diff
@@ -0,0 +1,118 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package datadogclientimpl
+
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+import (
+	"embed"
+	"io"
+
+	"gopkg.in/zorkian/go-datadog-api.v2"
+
+	datadogclient "github.com/DataDog/datadog-agent/comp/autoscaling/datadogclient/def"
+	"github.com/DataDog/datadog-agent/comp/core/status"
+)
+
+// statusProvider provides the functionality to populate the status output
+type statusProvider struct {
```

Comment:
This file is missing unit tests 😄 

Would be nice to have some tests to ensure the status output works as expected

Reply:
added